nntrainer/layers/common_properties.h

```diff
@@ -487,6 +508,38 @@ class Activation final : public EnumProperty<ActivationTypeInfo> {
   static constexpr const char *key = "activation";
 };
 
+/**
+ * @brief     Enumeration of pooling type
+ */
+struct PoolingTypeInfo {
+  /**
+   * @brief   Pooling operation type class
+   */
+  enum class Enum {
+    max = 0,
+    average = 1,
+    global_max = 2,
+    global_average = 3,
+    unknown = 4
+  };
+  static constexpr std::initializer_list<Enum> EnumList = {
+    Enum::max, Enum::average, Enum::global_max, Enum::global_average,
+    Enum::unknown};
+
+  static constexpr const char *EnumStr[] = {"max", "average", "global_max",
+                                            "global_average", "unknown"};
```

Comment:
Not sure with C++ grammar, but is it possible to express?:
```
{ [max] = "max", [average] = "average", ... };
```
This will ensure the correctness of future extensions.

Reply:
For now, it is necessary for the enum values to start with 0, and be contiguous.
Later, this can be replaced with a map/unordered_map and then the above can be supported.